mesa/experimental/jupyter_viz.py

```diff
@@ -59,107 +59,102 @@ def make_model():
     def handle_change_model_params(name: str, value: any):
         set_model_parameters({**model_parameters, name: value})
 
+    @solara.component
+    def ColorCard(title, color, type="Grid"):
+        with rv.Card(
+            style_=f"background-color: {color}; width: 100%; height: 100%"
+        ) as main:
+            rv.CardTitle(children=[title])
+            if type == "Grid":
+                if space_drawer == "default":
+                    # draw with the default implementation
+                    make_space(model, agent_portrayal)
+                elif space_drawer:
+                    # if specified, draw agent space with an alternate renderer
+                    space_drawer(model, agent_portrayal)
+            elif type == "Measure":
+                for measure in measures:
+                    if callable(measure):
+                        # Is a custom object
+                        measure(model)
+                    else:
+                        make_plot(model, measure)
+        return main
+
     # 3. Set up UI
-    solara.Markdown(name)
-    UserInputs(user_params, on_change=handle_change_model_params)
-    ModelController(model, play_interval, current_step, set_current_step, reset_counter)
-
-    with solara.GridFixed(columns=2):
-        # 4. Space
-        if space_drawer == "default":
-            # draw with the default implementation
-            make_space(model, agent_portrayal)
-        elif space_drawer:
-            # if specified, draw agent space with an alternate renderer
-            space_drawer(model, agent_portrayal)
-        # otherwise, do nothing (do not draw space)
 
-        # 5. Plots
-        for measure in measures:
-            if callable(measure):
-                # Is a custom object
-                measure(model)
-            else:
-                make_plot(model, measure)
+    with solara.AppBar():
+        solara.AppBarTitle(name)
+
+    grid_layout_initial = [
+        {"h": 12, "i": "0", "moved": False, "w": 5, "x": 0, "y": 0},
+        {"h": 12, "i": "1", "moved": False, "w": 5, "x": 7, "y": 0},
+    ]
+
+    colors = "white white".split()
+
+    # we need to store the state of the grid_layout ourselves, otherwise it will 'reset'
+    # each time we change resizable or draggable
+    grid_layout, set_grid_layout = solara.use_state(grid_layout_initial)
+
+    # render layout and plot
+
+    if "ipykernel" in sys.argv[0]:  # jupyter
```

Comment:
Can you split the following if-else into 2 components each defined in a function?

Reply:
component forces re-render of Cards , for now I have moved these to two functions